hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -2002,4 +2005,27 @@ public void fileCacheCompleted(Path filePath, long size) {
     regionCachedSizeMap.merge(regionName, size, (oldpf, fileSize) -> oldpf + fileSize);
   }
 
+  @Override
+  public Optional<Map<String, Integer>> uncacheStaleBlocks(HRegionServer server) {
+    Map<String, Integer> evictedFilesWithStaleBlocks = new HashMap<>();
+
+    fullyCachedFiles.forEach((fileName, value) -> {
+      int blocksEvicted;
+      try {
+        if (!server.getRegionByEncodedName(value.getFirst()).isAvailable()) {
+          blocksEvicted = evictBlocksByHfileName(fileName);
+        } else {
+          blocksEvicted = 0;
+        }
+      } catch (NotServingRegionException nsre) {
```

Comment:
Is this the normal path? In the above if condition, we test isAvailable, so why here we could still get a NotServingRegionException?